services/QuillDiagnostic/app/components/shared/uploadOptimalResponses.jsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react'
+import XLSX from 'xlsx'
+
+export default class UploadOptimalResponses extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { responses: [], }
+
+    this.handleChangeFile = this.handleChangeFile.bind(this)
+    this.submitResponses = this.submitResponses.bind(this)
+  }
+
+  handleChangeFile(file) {
+    const fileReader = new FileReader();
+    fileReader.onload = (e) => {
+      const data = new Uint8Array(e.target.result);
+      const workbook = XLSX.read(data, { type: 'array', });
+      const sheet = workbook.Sheets[workbook.SheetNames[0]]
+      const responses = Object.values(sheet).map(value => value.v).filter(Boolean)
+      this.setState({ responses: responses.slice(1), })
```

Comment:
Why are we taking `.slice(1)` here?  Again, I presume that it's because we're expecting the prompt as the first line and want to skip it, but putting a comment in to explain it might be helpful.